src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,24 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
     private final Executor executor;
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> cache
+                .computeIfAbsent(userId, this::getUserDataById), executor);
+    }
+
+    private UserData getUserDataById(String id) {
+        return new UserData(id, "Details for " + id);
```

Comment:
we should also have thread sleep here, as per the task requirements, to differentiate cases when we go to DB, and when we get the data from cache (which is faster)

Reply:
Sorry, I forgot about this.